native/core/src/parquet/read/values.rs

```diff
@@ -285,6 +259,59 @@ impl PlainDecoding for Int32DateType {
     }
 }
 
+impl PlainDecoding for Int32TimestampMicrosType {
+    #[inline]
+    fn decode(src: &mut PlainDecoderInner, dst: &mut ParquetMutableVector, num: usize) {
+        let src_data = &src.data;
+        let byte_width = src.desc.type_length() as usize;
+        let num_bytes = byte_width * num;
+
+        {
+            let mut offset = src.offset;
+            for _ in 0..num {
+                let v = &src_data[offset..offset + byte_width] as *const [u8] as *const u8
```

Comment:
Is there a reason why we have to use `as *const [u8] as *const u8 as *const i32` rather than `.as_ptr() as *const i32`?

Reply:
updated